cpp/diffraction/StaticStructureFactorDebye.cc

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2010-2020 The Regents of the University of Michigan
+// This file is from the freud project, released under the BSD 3-Clause License.
+
+#include <algorithm>
+#include <cmath>
+#include <limits>
+#include <stdexcept>
+
+#include "Box.h"
+#include "ManagedArray.h"
+#include "NeighborComputeFunctional.h"
+#include "NeighborQuery.h"
+#include "RDF.h"
+#include "StaticStructureFactorDebye.h"
+#include "utils.h"
+
+/*! \file StaticStructureFactorDebye.cc
+    \brief Routines for computing static structure factors.
+*/
+
+namespace freud { namespace diffraction {
+
+StaticStructureFactorDebye::StaticStructureFactorDebye(unsigned int bins, float k_max, float k_min)
+{
+    if (bins == 0)
+    {
+        throw std::invalid_argument("StaticStructureFactorDebye requires a nonzero number of bins.");
+    }
+    if (k_max <= 0)
+    {
+        throw std::invalid_argument("StaticStructureFactorDebye requires k_max to be positive.");
+    }
+    if (k_max <= k_min)
+    {
+        throw std::invalid_argument("StaticStructureFactorDebye requires that k_max must be greater than k_min.");
+    }
+
+    // Construct the Histogram object that will be used to track the structure factor
+    auto axes
+        = S_kHistogram::Axes {std::make_shared<util::RegularAxis>(bins, k_min, k_max)};
+    m_histogram = S_kHistogram(axes);
+    m_local_histograms = S_kHistogram::ThreadLocalHistogram(m_histogram);
+    m_min_valid_k = std::numeric_limits<float>::infinity();
+    m_structure_factor.prepare(bins);
+}
+
+void StaticStructureFactorDebye::accumulate(const freud::locality::NeighborQuery* neighbor_query,
```

Comment:
there are some places in this function where we use `float` and other places where we use `double`. @bdice What are freud's rules about when to use single vs double precision?

Reply:
@tommy-waltmann The rules are generally "use `float` where possible, and `double` where precision requires it." This is discussed further in #464. Also see the comment thread below: https://github.com/glotzerlab/freud/pull/660/files#r707543882

Resolving this.